### PR TITLE
add limit for transaction pool. print out error for debugging

### PR DESCRIPTION
**Pull request by mikedoan**
add limit for transaction pool. print out error for debugging

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -48,6 +48,11 @@ const (
 	NodeLeader                         // Node is the leader of some shard.
 )
 
+const (
+	// TxPoolLimit is the limit of transaction pool.
+	TxPoolLimit = 20000
```
Comment (LeoHChen) [inline node/node.go]: Can we make it bigger?  what's the proper number?

File: node/node.go [hunk 2]
```diff
@@ -223,8 +228,14 @@ func (node *Node) Beaconchain() *core.BlockChain {
 func (node *Node) addPendingTransactions(newTxs types.Transactions) {
 	node.pendingTxMutex.Lock()
 	node.pendingTransactions = append(node.pendingTransactions, newTxs...)
+	// If length of pendingTransactions is greater than TxPoolLimit then by greedy take the TxPoolLimit recent transactions.
+	if len(node.pendingTransactions) > TxPoolLimit {
+		utils.GetLogInstance().Error("Got more transactions than expected and this could caused OOM", "num", len(newTxs), "totalPending", len(node.pendingTransactions))
```
Comment (LeoHChen) [inline node/node.go]: Is this an error or warning?

Comment (rlan35) [reply node/node.go]: should be warning.

File: node/node.go (hunk 1 above)
Comment (rlan35) [reply node/node.go]: one tx is around 250-500 bytes. we should be able to do 200000

Comment (mikedoan) [reply node/node.go]: 1/ 20k TPS is a record for one shard. i don't think we would expect that.
2/ the goal of this is to find the culprit of OOM - If this really reaches 20k there is something wrong for example the finality is too long which, going forward, we need to look into.

Comment (mikedoan) [reply node/node.go]: discord discussed

File: node/node.go (hunk 2 above)
Comment (mikedoan) [reply node/node.go]: fixed

**Pull request merged**